specs/binding-kafka.spec/src/main/resources/META-INF/zilla/kafka.idl

```diff
@@ -241,25 +242,52 @@ scope kafka
             KafkaAckMode ackMode = IN_SYNC_REPLICAS;
         }
 
-        struct KafkaMergedDataEx
+        union KafkaMergedDataEx switch (uint8)
+        {
+            case 252: kafka::stream::KafkaMergedConsumerDataEx consumer;
+            case 1: kafka::stream::KafkaMergedFetchDataEx fetch;
+            case 0: kafka::stream::KafkaMergedProduceDataEx produce;
+        }
+
+        struct KafkaMergedProduceDataEx
         {
             int32 deferred = 0;         // INIT only (TODO: move to DATA frame)
             int64 timestamp = 0;        // INIT only
-            int64 filters = -1;         // INIT only
             KafkaOffset partition;      // INIT only
             KafkaOffset[] progress;     // INIT only
             KafkaKey key;               // INIT only
             KafkaKey hashKey;           // INIT only
+            KafkaHeader[] headers;      // INIT + FIN (produce), INIT only (fetch)
+        }
+
+        struct KafkaMergedFetchDataEx
+        {
+            int32 deferred = 0;         // INIT only (TODO: move to DATA frame)
+            int64 timestamp = 0;        // INIT only
+            int64 filters = -1;         // INIT only
+            KafkaOffset partition;      // INIT only
+            KafkaOffset[] progress;     // INIT only
+            KafkaKey key;               // INIT only
             KafkaDelta delta;           // INIT + FIN
             KafkaHeader[] headers;      // INIT + FIN (produce), INIT only (fetch)
         }
 
+        struct KafkaMergedConsumerDataEx
+        {
+            KafkaOffset progress;
```

Comment:
`progress` or `partition`? (actual question)

Also, do we need to be able to commit offsets for multiple partitions at once here? i.e. do we need an array?